kornia/core/check.py

```diff
@@ -102,7 +102,7 @@ def KORNIA_UNWRAP(maybe_obj: object, typ: Any) -> Any:
 T = TypeVar('T', bound=type)
 
 
-def KORNIA_CHECK_TYPE(x: object, typ: T, msg: str | None = None) -> TypeGuard[T]:
+def KORNIA_CHECK_TYPE(x: object, typ: T | tuple[T, ...], msg: str | None = None) -> TypeGuard[T]:
```

Comment:
It probably won't work with for the typeguard (mypy), but I don't know how it could be done either 😄

Reply:
Is it not passing mypy?

Reply:
example:

```python
from typing import TypeVar, TypeGuard, Type


def myf(a: int | str) -> int:
    return 1


T = TypeVar('T')


def KORNIA_CHECK_TYPE(x: object, typ: Type[T] | tuple[Type[T], ...]) -> TypeGuard[T]:
    if not isinstance(x, typ):
        raise
    return True


var_A = True
if KORNIA_CHECK_TYPE(var_A, (int, str)):
    print(myf(var_A))  # L19: not guard properly the typing
if KORNIA_CHECK_TYPE(var_A, int):
    print(myf(var_A))  # L21: works fine
if isinstance(var_A, (int, str)):
    print(myf(var_A))  # L24 works fine


var_B = 1
if KORNIA_CHECK_TYPE(var_B, (int, str)):
    print(myf(var_B))  # L28: not guard properly the typing
if isinstance(var_B, (int, str)):
    print(myf(var_B))  # L30: works fine
```

mypy output:
```bash
$ mypy t.py 
t.py:19: error: Argument 1 to "myf" has incompatible type "object"; expected "Union[int, str]"  [arg-type]
t.py:28: error: Argument 1 to "myf" has incompatible type "object"; expected "Union[int, str]"  [arg-type]
```


The problem here is something like this: the `typ` tuple is a tuple of `type_A`, `type_B`, ..., `type_N` -- so how we can properly type this for the type guard can guarantee that the `x` object is one of these `N` types

Reply:
i think we can leave it as it is, just add a TODO comment. I am not sure if there is a solution for this typing, anyway some time would have to be put on top of this to find a solution